numpy/core/src/multiarray/item_selection.c

```diff
@@ -1204,25 +1241,35 @@ _new_sortlike(PyArrayObject *op, int axis, PyArray_SortFunc *sort,
         }
 
         if (needcopy) {
-            copyswapn(it->dataptr, astride, buffer, elsize, N, swap, op);
+            char *args[2] = {buffer, it->dataptr};
+            npy_intp strides[2] = {elsize, astride};
+
+            if (NPY_UNLIKELY(from_cast_info.func(
```

Comment:
I don't think the UNLIKELY matters here since there is nothing within the if clause, but doesn't matter.